packages/svelte/src/internal/server/index.js

```diff
@@ -95,27 +95,30 @@ export let on_destroy = [];
 /**
  * Only available on the server and when compiling with the `server` option.
  * Takes a component and returns an object with `body` and `head` properties on it, which you can use to populate the HTML when server-rendering your app.
- * @template {Record<string, any>} Props
- * @param {import('svelte').Component<Props> | import('svelte').ComponentType<import('svelte').SvelteComponent<Props>>} component
- * @param {{ props: Omit<Props, '$$slots' | '$$events'>; context?: Map<any, any> }} options
+ * @template {import('svelte').Component | import('svelte').ComponentType<import('svelte').SvelteComponent<any>>} Component
+ * @template {import("svelte").ComponentProps<import('../types.js').ComponentOrSvelteComponent<Component>>} [Props=import("svelte").ComponentProps<import('../types.js').ComponentOrSvelteComponent<Component>>]
+ *
+ * @param {Component} component
+ * @param {import('../types.js').AllOptionalProps<NoInfer<Props>> extends true ? [{ props: Omit<NoInfer<Props>, '$$slots' | '$$events'>; context?: Map<any, any> }] | [] : [{ props: Omit<NoInfer<Props>, '$$slots' | '$$events'>; context?: Map<any, any> }]} options
  * @returns {import('#server').RenderOutput}
  */
-export function render(component, options) {
+export function render(component, ...options) {
+	const option = options[0];
```

Comment:
Is there any way to avoid this extra runtime overhead? Can we just keep doing something simpler here if we're also updating the types to be more complete?

Reply:
Not that I know of because it needs to be a rest parameter for the trick to work.

We could maybe try with overloads even tho I don't properly love them